activesupport/lib/active_support/number_helper/number_converter.rb

```diff
@@ -173,8 +175,13 @@ def default_value(key)
           key.split(".").reduce(DEFAULTS) { |defaults, k| defaults[k.to_sym] }
         end
 
-        def valid_float?
-          Float(number, exception: false)
+        def valid_bigdecimal
+          case number
+          when Float, Rational
+            number.to_d(0)
+          else
+            BigDecimal(number, exception: false)
+          end
```

Comment:
This is the big fix. The other changes are because this method was renamed since its return type was changed.

Reply:
Would there be a problem with something like this?
```suggestion
          case number
          when Float, Rational
            number.to_d(0)
          else
            number.to_d rescue nil
          end
```

I recently ran into a problem where I'm using a ["lazy object"](https://github.com/HornsAndHooves/lazy_object) that gets passed to a number helper. Previously this worked fine, but now `BigDecimal()` is not converting the object, because it sees the number as a `LazyObject` instead of the nested object being delegated to.

If my object was sent `to_d`, it would behave as expected. Interestingly, `Float(lazy_object)` works, but `BigDecimal(lazy_object)` does not.

Reply:
Can you define `to_str` on you object and try to see if it works? As per documentation https://ruby-doc.org/stdlib-3.1.1/libdoc/bigdecimal/rdoc/Kernel.html#method-i-BigDecimal

Reply:
Since the lazy object wraps an Integer, I would need to define to_str on Integer which I would prefer not to do. I will just change my code to not give the number helper a lazy object, since I'm sure this is not going to impact many other people.